persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -67,6 +67,10 @@ abstract class AbstractPersistentEntityRegistry(system: ActorSystem, injector: I
   }
 
   private val registeredTypeNames = new ConcurrentHashMap[String, Class[_]]()
+  private val reverseRegister: (Class[_]) => Option[String] = { entityClass =>
+    import scala.collection.JavaConversions.mapAsScalaConcurrentMap
+    registeredTypeNames.filter { case (name, className) => className == entityClass }.keys.headOption
+  }
```

Comment:
Can you explain the race condition?  There's no reason updating the two maps needs to be atomic, or exclusive.  Both should do a `putIfAbsent`, and check the result to make sure it was absent, in either case if it wasn't, an exception should be thrown, and that exception will ultimately prevent the entire service from starting up, which will mean it doesn't matter what the state of the two maps is.

Also, do you know what the cost of this is?  Does `filter` on a Scala `ConcurrentMap` return a copy or a view?  Does `keys` return a copy or a view?